### ISSUE TITLE
PPO rewards have been oscillating

### ISSUE THREAD

**Issue by HaoningJiang-space** [Dyllon-Preston/trajopt-grpo#2]
The reward of PPO is oscillating 
-80.79838562011719
-84.19493865966797
-76.98068237304688
-76.46812438964844
-83.26513671875
-85.55415344238281
-79.5936279296875
-94.66239929199219
-87.97920227050781
-83.26356506347656
-77.0241928100586
-85.18900299072266
-83.54051971435547
-83.26643371582031
-82.36634826660156
-77.09642791748047
-81.04037475585938
-74.20174407958984
-81.23965454101562
-79.57539367675781
-83.18138885498047
-85.99605560302734
-79.44956970214844
-81.4971923828125
-80.61309051513672
-77.08263397216797
-91.0831069946289
-80.84777069091797
-92.01506805419922
-78.21715545654297
-88.19890594482422
-85.69939422607422
-83.11412048339844
-83.23877716064453
-81.7782211303711
-88.1987075805664
-82.27923583984375
-83.34623718261719
-74.82390594482422
-82.53356170654297
-78.02587890625
-77.88241577148438
-76.30950927734375
-75.6435546875
-82.60550689697266
-81.06881713867188
-85.25167846679688
-75.79850769042969
-88.15392303466797
-77.85427856445312
-78.83607482910156
-85.451904296875
-82.23912811279297
-85.08677673339844
-73.30461120605469
-86.66816711425781
-83.44782257080078
-79.9526138305664
-94.41712188720703
-84.59699249267578
-83.61966705322266
-85.61656951904297
-77.9390869140625
-78.1021957397461
-74.14268493652344
-93.63651275634766
-86.52928924560547
-85.9515609741211
-94.78679656982422
-85.3252182006836

**Comment by HaoningJiang-space** [comment]
> The reward of PPO is oscillating -80.79838562011719 -84.19493865966797 -76.98068237304688 -76.46812438964844 -83.26513671875 -85.55415344238281 -79.5936279296875 -94.66239929199219 -87.97920227050781 -83.26356506347656 -77.0241928100586 -85.18900299072266 -83.54051971435547 -83.26643371582031 -82.36634826660156 -77.09642791748047 -81.04037475585938 -74.20174407958984 -81.23965454101562 -79.57539367675781 -83.18138885498047 -85.99605560302734 -79.44956970214844 -81.4971923828125 -80.61309051513672 -77.08263397216797 -91.0831069946289 -80.84777069091797 -92.01506805419922 -78.21715545654297 -88.19890594482422 -85.69939422607422 -83.11412048339844 -83.23877716064453 -81.7782211303711 -88.1987075805664 -82.27923583984375 -83.34623718261719 -74.82390594482422 -82.53356170654297 -78.02587890625 -77.88241577148438 -76.30950927734375 -75.6435546875 -82.60550689697266 -81.06881713867188 -85.25167846679688 -75.79850769042969 -88.15392303466797 -77.85427856445312 -78.83607482910156 -85.451904296875 -82.23912811279297 -85.08677673339844 -73.30461120605469 -86.66816711425781 -83.44782257080078 -79.9526138305664 -94.41712188720703 -84.59699249267578 -83.61966705322266 -85.61656951904297 -77.9390869140625 -78.1021957397461 -74.14268493652344 -93.63651275634766 -86.52928924560547 -85.9515609741211 -94.78679656982422 -85.3252182006836

In script/cartpole_nn_ppo

**Comment by Dyllon-Preston** [comment]
The default parameters should be able to converge if you just run `scripts/cartpole_nn_ppo`.

However, just in case, I tweaked the cartpole environment for better convergence and just pushed to main. If you pull the code and run `script/cartpole_nn_ppo`, you should see a result that looks something like this:

![CartPole PPO Simulation](https://github.com/Dyllon-Preston/trajopt-grpo/blob/main/reports/CartPole/cartpole_nn_ppo/001/simulation.gif?raw=true)

I did some reward engineering to improve the gradients that point towards balancing the pendulum. I also reduced the size of the policy model to prevent overfitting.

Small oscillations while training are common as the policy learns. However, over many epochs, you should see a general improvement in the average total reward per episode.
